src/twisted/conch/ssh/session.py

```diff
@@ -91,6 +92,33 @@ def request_pty_req(self, data):
         else:
             return 1
 
+    def request_env(self, data):
+        """
+        Process a request to pass an environment variable.
+
+        @param data: The environment variable name and value, each encoded
+            as an SSH protocol string and concatenated.
+        @type data: L{bytes}
+        @return: A true value if the request to pass this environment
+            variable was accepted, otherwise a false value.
+        """
+        if not self.sessionSetEnv:
+            try:
+                self.sessionSetEnv = ISessionSetEnv(self.avatar)
```

Comment:
A better way to address the lack of type information than catching `TypeError` (which may be raised by a great many things in Python's runtime or standard library) is to do `ISessionSetEnv(self.avatar, None)`.